app/utils/Assets.scala

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import java.net.URL
+import controllers.CachedAssets.hashedPaths
+import play.api.Play
+import org.apache.commons.io.IOUtils
+
+object Assets {
+
+  def inlineResource(path: String) = {
+    val url = AssetFinder("public/" + hashedPaths.getOrElse(path, path))
+    IOUtils.toString(url)
+  }
+
+  val svgSprite = inlineResource("images/svg-sprite.svg")
+}
+
+object AssetFinder {
+  def apply(assetPath: String): URL = {
```

Comment:
I would move this logic into your main Singleton object `Assets, and wrap it into a method called something like `Assets.find`